### PR TITLE
Allow CREATE requests for ShootStates,BackupEntries during control plane migration

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security control-plane-migration
/kind enhancement

**What this PR does / why we need it**:
The `SeedRestriction` admission handler needs to be aware of the control plane migration use-case and allow the gardenlets of the source seeds to properly perform their migration flow:

```
task "Ensuring that ShootState exists" failed: retry failed with context deadline exceeded, last error: admission webhook "seed-restriction.gardener.cloud" denied the request: object does not belong to seed "aws"
```

**Special notes for your reviewer**:
/invite @timuthy @plkokanov @kris94 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by plkokanov — DISMISSED** [review]

**Comment by plkokanov** [comment]
Generally lgtm
Just not sure how this is affecting the extensions crd deletion confirmation test :(

File: pkg/admissioncontroller/webhooks/admission/seedrestriction/admission.go [hunk 1]
```diff
@@ -179,7 +179,7 @@ func (h *handler) admitBackupEntry(ctx context.Context, seedName string, request
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	if resp := h.admit(seedName, backupEntry.Spec.SeedName); !resp.Allowed {
+	if resp := h.admit(seedName, backupEntry.Spec.SeedName, backupEntry.Status.SeedName); !resp.Allowed {
```
Comment (timuthy) [inline pkg/admissioncontroller/webhooks/admission/seedrestriction/admission.go]: I only see `update` calls for this use-case in the migration flow after having a quick look. Can you elaborate why we need to cover `BackupEntries` as well?

Comment (rfranzke) [reply pkg/admissioncontroller/webhooks/admission/seedrestriction/admission.go]: Indeed, it seems to not be needed. Reverted it.
/squash

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**